### PR TITLE
Add metrics from mgr:info on squid_object

### PR DESCRIPTION
**Pull request by Tofolcr**
This pull request introduces two features.

Firstly, it retrieves info from the cache_object and exposes all numeric values as metrics.

Secondly, it incorporates a label array with key:value pairs within types.counter. This addition aims to generate an extra metric (squid_info_service) with labels encompassing string values found in the info page. These labels include squid_version, service_name, start_time, Build_Info, among others.

All newly introduced metrics in this pull request commence with the prefix "squid_info_*."

I have meticulously tested the exporter using a Docker Compose setup, which includes one exporter, one squid, and one Prometheus. I am pleased to report that everything functions seamlessly.

I welcome all comments and reviews to further enhance the proposed changes.

Fixes [#66](https://github.com/boynux/squid-exporter/issues/66)

### CODE REVIEW

File: collector/client.go [hunk 1]
```diff
@@ -231,3 +265,76 @@ func decodeServiceTimeStrings(line string) (types.Counter, error) {
...
+				infoCounter.Key = key
+				infoCounter.VarLabels = append(infoCounter.VarLabels, infoVarLabel)
+				return infoCounter, nil
+			} else if key == "Start_Time" || key == "Current_Time" {
+				return types.Counter{}, nil
+			}
+
+			// Remove additional formating string
+			if slices := strings.Split(value, " "); len(slices) > 0 {
+				if slices[0] == "5min:" {
```
Comment (boynux) [inline collector/client.go:308]: can we also collect 60m counters?

File: collector/client.go [hunk 2]
```diff
@@ -155,6 +156,39 @@ func (c *CacheObjectClient) GetServiceTimes() (types.Counters, error) {
...
+
+	lines := make(chan string)
+	go readLines(reader, lines)
+
+	var infoVarLabels types.Counter
+	infoVarLabels.Key = "squid_info"
+	infoVarLabels.Value = 1
+
+	for line := range lines {
+		i, err := decodeInfoStrings(line)
```
[the remark this answers is not in the document]
Comment (Tofolcr) [reply collector/client.go]: changed var name i to dis

File: collector/client.go (hunk 1 above)
Comment (Tofolcr) [reply collector/client.go]: yes, I push the changes in order to collect 5m and 60m counters

File: collector/client.go [hunk 3]
```diff
@@ -231,3 +265,76 @@ func decodeServiceTimeStrings(line string) (types.Counter, error) {
 
 	return types.Counter{}, errors.New("service times - could not parse line: " + line)
 }
+
+func decodeInfoStrings(line string) (types.Counter, error) {
+	if strings.HasSuffix(line, ":\n") { // A header line isn't a metric
+		return types.Counter{}, nil
+	}
+
+	if equal := strings.Index(line, ":"); equal >= 0 {
```
[the remark this answers is not in the document]
Comment (Tofolcr) [reply collector/client.go]: changed de var name equal to idx

File: collector/client.go [hunk 4]
```diff
@@ -231,3 +265,76 @@ func decodeServiceTimeStrings(line string) (types.Counter, error) {
...
+			value = strings.Replace(value, ",", "", -1)
+
+			if i, err := strconv.ParseFloat(value, 64); err == nil {
+				return types.Counter{Key: key, Value: i}, nil
+			}
+		}
+	} else {
+		lineTrimed := strings.TrimSpace(line[:])
+
+		if equal := strings.Index(lineTrimed, " "); equal >= 0 {
```
[the remark this answers is not in the document]
Comment (Tofolcr) [reply collector/client.go]: changed de var name equal to idx

File: collector/client.go [hunk 5]
```diff
@@ -231,3 +265,76 @@ func decodeServiceTimeStrings(line string) (types.Counter, error) {
 
 	return types.Counter{}, errors.New("service times - could not parse line: " + line)
 }
+
+func decodeInfoStrings(line string) (types.Counter, error) {
```
[the remark this answers is not in the document]
Comment (Tofolcr) [reply collector/client.go:286]: In the last commit, I added some comments, let me know if it's enough or need to add more.

[the remark this answers is not in the document]
Comment (boynux) [reply collector/client.go:286]: Sounds good!

**Review by boynux — APPROVED** [review]

**Pull request merged**